src/notion-provider.ts

```diff
@@ -34,81 +50,78 @@ function NotionProvider(this: any, options: NotionProviderOptions) {
         cmd: {
           list: {
 	    action: async function(this: any, entize: any, msg: any) {
-	      let limit
-	      if(msg.q){
-                limit = msg.q.limit$ || 100 // The maximum number of results to display per page
-	      }
-	      else{
-	       limit = 100
+	      let q = msg.q || {}
+
+	      const config = {
+		body: { ...q }
 	      }
-	      const options = {
-	        method: 'POST',
-		headers: {
-		  'Authorization' : `Bearer ${this.shared.authTok}`,
-		  'Accept': 'application/json',
-		  'Notion-Version': '2022-02-22',
-		  'Content-Type': 'application/json'
-		},
-		body: JSON.stringify({page_size: limit})
-	      };
 	      // see https://developers.notion.com/reference/post-search for usage
-	      let res = await fetch('https://api.notion.com/v1/search', options)
-	      let obj = await res.json()
-	      let list = obj.results.filter(v => v.object == 'page')
-				    .map(v => entize(v))
+	      let res: any = await postJSON('https://api.notion.com/v1/search', makeConfig(config))
+	      
+	      let list = res.results.filter((v: any) => v.object == 'page')
+				    .map((v: any) => entize(v))
 	      return list
             }
           },
 
           load: {
             action: async function(this: any, entize: any, msg: any) {
               let id = msg.q.id
-	      const options = {
-	        method: 'GET',
-		headers: {
-		  'Authorization': `Bearer ${this.shared.authTok}`,
-		  'Accept': 'application/json',
-		  'Notion-Version': '2022-02-22',
-		  'Content-Type': 'application/json'
-		}
-	      }
-	      let res = await fetch(`https://api.notion.com/v1/pages/${id}`, options)
-              let obj = await res.json()
-	      return entize(obj)
+
+              null == id ? this.fail('invalid_id') : null
+
+	      // let res = await getJSON(`https://api.notion.com/v1/pages/${id}`, makeConfig())
+	      let res = await this.shared.sdk.pages.retrieve({page_id: id})
+
+	      return entize(res)
             }
           },
                     
 	  save: {
 	    action: async function(this: any, entize: any, msg: any) {
 	      let ent = msg.ent
 	      let id = ent.id
-	      let properties = ent.properties
+	      let page = ent.page
 	      let obj
 	      try{
 	        await this.shared.sdk.pages.update({page_id: id, properties: ent.properties})
 		// obj = await this.entity('provider/notion/page').load$(id) // a fix to get all the properties
 	      }
-	      catch(err){
+	      catch(err: any) {
 	        if(err.status >= 400 && err.status < 500){
```

Comment:
does the sdk return errors with status codes?

Reply:
since we are using `fetch` we handle this by throwing an Error. `postJSON` from `makeUtils` will throw it

Reply:
so yeah - that is cleaned up